### ISSUE TITLE
VictoriaLogs docs: Further clarify when to include fields in streams

### ISSUE THREAD

**Issue by Fusl** [VictoriaMetrics/VictoriaMetrics#7247]
### Is your question request related to a specific component?

VictoriaLogs

### Describe the question in detail

For us, the documentation for VictoriaLogs is not clear enough on whether to include certain fields in log streams. Our application generates about **30-40k unique log streams per day** or 200k per week and our **retention period is configured to be 3 months** (may be changed to 1 month later on as we also store a history of logs in S3). Due to the uncertainty of the performance of log streams, we decided to not use this feature for now. We intend on keeping it this way for simplciity reasons but it is unclear to us whether its use is recommended based on the following information.

**Potential Performance Improvements or Degradation**
Does the use of log streams during ingestion and log processing improve the performance when we **never** make use of log streams during querying (never query based on fields included in log streams)?

The documentation mentions potential "Reduced disk space usage" but does this still hold true when the same application producing the same type of logs is only being grouped into log streams based on the server they're running on?

**High cardinality**
The documentation warns about never associating high-cardinality fields with log streams. About how much is considered too much here and does the performance of VictoriaLogs degrade with the amount of unique log streams existing?

To clarify on this question, here's a real world example based on a single server application we run on our infrastructure:

**Container Hostname**
Our server application runs on hundreds of servers inside containers that often get replaced with different servers for load balancing reasons. The hostname is a randomly generated string. Our churn rate of the hostname field is therefore about **2500 values per day**.

**Machine Hostname**
This being the host name of the underlying container host machine has about the same churn rate of the container hostname with about **2000 values per day**.

**Application Version**
The version of our server application may change several times per day (lets assume a deployment every hour for 12 hours straight), lets go with a churn rate of **12 values per day**.

**Deployment Group**
Our server cluster runs a blue/green deployment group and we sometimes promote the standby group with the version running on the active group so that the standby group can become the active group with the latest version. This happens about 2-3 times per week, effectively a churn rate of **0.5 values per day**.

**Application Configuration Type**
We utilize a split group of user-facing and backend-facing groups of servers, essentially a 1:1 copy of our application running on the same version but different hostnames. This effectively **doubles the hostname values** churn rate from 3000 to 6000 per day and would be another field in our log group.

**Overall Churn Rate**
Calculating all of this information together, our application would generate about **30000 to 40000 unique log streams per day** or **200k unique log streams per week**.

### Troubleshooting docs

- [ ] General - https://docs.victoriametrics.com/troubleshooting/
- [ ] vmagent - https://docs.victoriametrics.com/vmagent/#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert/#troubleshooting

**Comment by Fusl** [comment]
TL;DR below

I've been doing some testing to answer my own question, here's my understanding from my observations:

I have tested with slowly ingesting logs across exactly 16,777,216 unique streams while ingesting high throughput logs for the same 10,000 streams. The performance of ingesting those logs barely dropped off (maybe 1-2% reduction) and the only time we see short, temporary slowdowns is during creation if new streams.

VictoriaLogs groups streams into the daily partitions. It doesn't seem to be a problem at all if we generate billions of streams over a longer time period (multiple weeks) as long as the number of streams within the daily partitions doesn't grow past a few millions.

We've now rolled out changed into production that separate logs into streams with a daily churn rate of unique log streams ~45,000 and have not noticed any performance impact.

TL;DR: We should at the very least add notes in the docs for the following info:
* Streams get grouped into daily partitions.
* To calculate the **absolute worst case scenario** of streams per day, one can use the basic formula of `num_field1_values * num_field2_value * num_field3_values`. For example, when using `_stream_fields=_HOSTNAME,_SYSTEMD_UNIT,PRIORITY`, if the amount of hosts is 50, the amount of systemd units is 200, and the amount of syslog priorities being 192, this would result in `50 * 200 * 192 =` **1,920,000** potential unique streams.
* It's possible to create millions of streams, but to reduce the performance impact, the amount of **actively** ingested streams should be kept within the thousands streams range.
